distribution/openrct2.d.ts

```diff
@@ -387,96 +585,211 @@ declare global {
+    /**
+     * A tile element.
+     */
     interface BaseTileElement {
+        /** The tile type. */
         type: TileElementType;
+        /** The base height of the tile for collision. */
         baseHeight: number;
+        /** The top height of the tile for collision. */
         clearanceHeight: number;
-        isHidden: boolean; /** Take caution when changing this field, it may invalidate TileElements you have stored in your script. */
+        /** Indicates if the tile is hidden. Take caution when changing this field, it may invalidate 
+         * TileElements you have stored in your plugin. 
+         */
+        isHidden: boolean;
     }
 
+    /**
+     * A surface element.
+     */
     interface SurfaceElement extends BaseTileElement {
+        /** TODO what does the number mean? */
         slope: number;
+        /** A number indicating the surface style. */
         surfaceStyle: number;
+        /** A number indicating the edge style (blended or not?) TODO: verify */
         edgeStyle: number;
+        /** The surface height of the water. */
         waterHeight: number;
+        /** The length of the grass from X to Y. TODO */
         grassLength: number;
+        /** A number from x to y indicating the ownership status. TODO */
         ownership: number;
+        /** A number indicating the location of park fences. */
         parkFences: number;
 
+        /** True if the surface is owned by the park, false otherwise. */
         readonly hasOwnership: boolean;
+        /** True if the surface has construction rights owned by the park, false otherwise. */
         readonly hasConstructionRights: boolean;
     }
 
+    /**
+     * A footpath element.
+     */
     interface FootpathElement extends BaseTileElement {
+        /** A number indicating the footpath type. TODO */
         footpathType: number;
+        /** A number indicating the connection points on the path. TODO */
         edgesAndCorners: number;
+        /** A number between x and y indicating the slope direction, or null for a flat path. TODO */
         slopeDirection: number | null;
+        /** True if the path has a crossing which is occupied (e.g. by a train), false otherwise. */
         isBlockedByVehicle: boolean;
+        /** TODO? Is this for better pathfinding? */
         isWide: boolean;
 
+        /** Indicates if the path is a queue. */
         isQueue: boolean;
+        /** A number between x and y indicating the queue banner direction, or null if no banner is present. TODO */
         queueBannerDirection: number | null;
+        /** TODO? */
         ride: number;
+        /** TODO? */
         station: number;
 
+        /** TODO? */
         addition: number | null;
+        /** TODO? */
         isAdditionBroken: boolean;
     }
 
+    /**
+     * A track element.
+     */
     interface TrackElement extends BaseTileElement {
+        /** A number indicating the track type. */
         trackType: number;
+        /** TODO? */
```

Comment:
multi tile track elements such as turns have a sequence number from 0 -> number of tiles that are occupied.